Safety/Src/safety_controller.cpp

```diff
@@ -0,0 +1,62 @@
+#include "safety_controller.hpp"
+#include "PWM.hpp"
+#include "stdlib.h"
+#include "stm32f0xx_hal_iwdg.h"
+
+
+
+//static Interchip_AtoS_Packet *dataRX;
+//static Interchip_StoA_Packet *dataTX;
+
+void safety_controller_init()
+{
+    
+    //dataTX = (Interchip_StoA_Packet*) malloc(sizeof(Interchip_StoA_Packet));
+    //dataRX = (Interchip_AtoS_Packet*) malloc(sizeof(Interchip_AtoS_Packet));
+    //Interchip_Init(dataTX, dataRX);
+}
+
+void safety_run(PWMChannel &pwm, PPMChannel &ppm)
+{
+    if(isSafetyManual())
+    {
+        for(int channel = 0; channel < 12; channel++)
+        {
+            //setPWMChannel(pwm, channel, dataRX->PWM[channel]);
+        }   
+    }
+    else
+    {
+        for(int channel = 0; channel < 12; channel++)
+        {
+            setPWMChannel(pwm, channel, getPPM(ppm, channel));
+        }
+    }
+    
+}
+
+
+/*
+* Sets an individual PWM channel to the desired output.
+*/
+void setPWMChannel(PWMChannel &pwm, int channel, int percentage)
```

Comment:
noted earlier, but all helper functions should be declared static and their prototypes should be at the top of the cpp file (so that they're file scope).